### PR TITLE
Upgrade SpriteEffect, remove limited-use effects 

### PR DESCRIPTION
**Pull request by reaperrr**
...and unhardcode `GainsExperience` levelup effect.

Originally started working on this because I stumbled over the latter.

In summary, this PR
- adds support for dynamically updating position and facing of `SpriteEffect` while the animation is running
- uses the above to replace the recently added `LaunchEffect` as well as `CrateEffect`
- unhardcodes (and disables by default) the levelup crate effect spawned by `GainsExperience` (so mods that aren't based on any official mod can use GainsExperience without a levelup crate effect)

### CODE REVIEW

**Comment by reaperrr** [comment]
> adds support for dynamically updating position and facing of SpriteEffect while the animation is running

This should also bring us a step closer to TS-style smoke particles.

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/DefineLevelupImageDefault.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class DefineLevelupImageDefault : UpdateRule
```
Comment (pchote) [inline OpenRA.Mods.Common/UpdateRules/Rules/20180923/DefineLevelupImageDefault.cs]: `DefineLevelUpImageDefault` (`U` should be capitalized)

File: OpenRA.Mods.Common/Effects/SpriteEffect.cs [hunk 2]
```diff
@@ -20,25 +21,44 @@ public class SpriteEffect : IEffect, ISpatiallyPartitionable
...
 		readonly bool scaleSizeWithZoom;
+		readonly bool staticPosition;
+		WPos pos;
 
+		// Facing is last on these overloads partially for backwards compatibility with previous main ctor revision
+		// and partially because most effects don't need it.
 		public SpriteEffect(WPos pos, World world, string image, string sequence, string palette, bool visibleThroughFog = false, bool scaleSizeWithZoom = false, int facing = 0)
+			: this(() => pos, () => facing, world, image, sequence, palette, visibleThroughFog, scaleSizeWithZoom)
+		{
+			staticPosition = true;
```
Comment (pchote) [inline OpenRA.Mods.Common/Effects/SpriteEffect.cs]: Evaluating `posFunc` has a negligible perf impact, so we could remove this and keep things simpler.

File: OpenRA.Mods.Common/Traits/Crates/CrateAction.cs [hunk 3]
```diff
@@ -20,8 +20,11 @@ public class CrateActionInfo : ITraitInfo
 		[Desc("Chance of getting this crate, assuming the collector is compatible.")]
 		public readonly int SelectionShares = 10;
 
-		[Desc("An animation defined in sequence yaml(s) to draw.")]
-		public readonly string Effect = null;
+		[Desc("Image containing the crate effect animation sequence.")]
+		public readonly string Image = "crate-effects";
+
+		[Desc("Animation sequence played when collected. Leave empty for no effect.")]
+		[SequenceReference("Image")] public readonly string Effect = null;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Crates/CrateAction.cs]: Can we rename this to `Sequence` for consistency with other traits?

**Review by pchote — DISMISSED** [review]

**Review by pchote — DISMISSED** [review]

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
Considering the lack of review bandwidth i'll take this without a second :+1:

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/2495c4f82cdaa0b9c6f229313b22d83ddd097529

**Pull request merged**